### ISSUE TITLE
Навчити MLP, XGBoost і LightGBM

### ISSUE THREAD

**Issue by marakonec** [DIIT-KIT-Hub/dda-mobile#30]
**Контекст.** Окрім baseline LR навчаємо ще три моделі:
- **MLP** (Keras) нейромережа: легко експортується у TFLite, добре «розкрутить» нелінійні залежності між features.
- **XGBoost** і **LightGBM** gradient boosting, який часто перемагає нейромережі на табличних даних з малим N. Експорт у TFLite через двійковий формат + ONNX, або просто реімплементація в Kotlin.
Усі чотири моделі дають порівняльну таблицю для статті: ми спробували весь стандартний набір, обрана модель ... .

**Що зробити.**
1. У `03-model-training.ipynb` додати три секції по одній на модель.
2. **MLP**: 2-3 шари по 32-64 нейрони, ReLU, Adam, MSE loss, EarlyStopping на val.
3. **XGBoost**: `XGBRegressor(n_estimators=300, max_depth=4, learning_rate=0.05)`.
4. **LightGBM**: `LGBMRegressor(n_estimators=300, max_depth=6, learning_rate=0.05)`.
5. Hyperparameter tuning через Optuna (50 trials, objective = MAE на val) для XGB і LGBM.
6. Записати усі метрики у DataFrame, експортувати у Markdown-таблицю.
7. Побудувати feature importance plots для XGB і LGBM (`plot_importance`).
8. Обрати best model (мінімум MAE на test) обґрунтувати у статті.

**Як зробити.** Скелет:
```python
import optuna, xgboost as xgb, lightgbm as lgb
from sklearn.metrics import mean_absolute_error
import tensorflow as tf

# ----- MLP -----
def make_mlp(input_dim):
    m = tf.keras.Sequential([
        tf.keras.layers.Input(shape=(input_dim,)),
        tf.keras.layers.Dense(64, activation="relu"),
        tf.keras.layers.Dense(32, activation="relu"),
        tf.keras.layers.Dense(1)
    ])
    m.compile(optimizer="adam", loss="mse", metrics=["mae"])
    return m

mlp = make_mlp(X_train.shape[1])
mlp.fit(X_train, y_train, validation_data=(X_val, y_val),
        epochs=200, batch_size=32,
        callbacks=[tf.keras.callbacks.EarlyStopping(patience=15, restore_best_weights=True)])

# ----- XGBoost з Optuna -----
def xgb_objective(trial):
    params = {
        "n_estimators": trial.suggest_int("n_estimators", 100, 500),
        "max_depth": trial.suggest_int("max_depth", 3, 8),
        "learning_rate": trial.suggest_float("lr", 0.01, 0.2, log=True),
        "subsample": trial.suggest_float("subsample", 0.6, 1.0),
    }
    m = xgb.XGBRegressor(**params, random_state=42).fit(X_train, y_train)
    return mean_absolute_error(y_val, m.predict(X_val))

study = optuna.create_study(direction="minimize")
study.optimize(xgb_objective, n_trials=50)
best_xgb = xgb.XGBRegressor(**study.best_params, random_state=42).fit(X_train, y_train)

# ----- LightGBM (аналогічно) -----
# ... (той самий патерн)

# ----- Зведена таблиця -----
results = pd.DataFrame([
    {"model": "Linear Regression", **eval_metrics(lr, X_test, y_test)},
    {"model": "MLP", **eval_metrics(mlp, X_test, y_test)},
    {"model": "XGBoost", **eval_metrics(best_xgb, X_test, y_test)},
    {"model": "LightGBM", **eval_metrics(best_lgb, X_test, y_test)},
])
print(results.to_markdown(index=False))
```

**Результат.** `docs/results/model-comparison.md` містить:
- Таблицю Model × MAE × RMSE × R² × Training time × Inference latency (estimate).
- Feature importance plots для XGB і LGBM (PNG у `docs/results/figures/`).
- Параграф Selected model з обґрунтуванням вибору і коротким аналізом, чому інші моделі гірші/кращі.
